Frontend/ui-customer/src/Flow.purs

```diff
@@ -983,11 +1010,31 @@ homeScreenFlow = do
               })
         if isMoreThan20Meters || cachedLocation == "" then do
           PlaceName placeDetails <- getPlaceName lat lon gateAddress
+          let currentLocationItem = Just (locationListStateObj{
+                              prefixImageUrl = fetchImage FF_ASSET "ny_ic_recent_search"
+                            , postfixImageUrl = fetchImage FF_ASSET "ny_ic_fav"
+                            , title = (fromMaybe "" ((split (Pattern ",") (placeDetails.formattedAddress)) !! 0))
+                            , subTitle = let formattedAddress = placeDetails.formattedAddress
+                                             index = fromMaybe 0 (indexOf (Pattern ",") formattedAddress)
+                                          in drop (index + 2) formattedAddress
+                            , placeId = placeDetails.placeId
+                            , lat = Just (if state.props.currentStage /= ConfirmingLocation then lat 
+                                        else if (state.props.isSource == Just true) then state.props.sourceLat
```

Comment:
`state.props.isSource == Just true` this looks very bad everywhere, good to replace it with `isJust` by removing `Just false` usecase